### PR TITLE
Restricting a resident to have only one active residency at a time

### LINKED ISSUE #425: A resident with active residency cannot be added to new residency
In the  `add a new residency` modal , the resident dropdown cannot have residents with active residency (i.e. residency does not have moveOut date). 
Also on submitting the form, a check on the backend to ensure that no user is added such that it has multiple active residencies

### PR DESCRIPTION
**Pull request by shailee-m**
closes #425

### CODE REVIEW

**Comment by brylie** [comment]
This code looks good, but I just want to reconsider the approach. I like that we are checking for 'active residency' by seeing if there is a `moveOut` date, which is consistent with other places.

That said, I think what we need to prevent here are **overlapping residencies**. I.e. residencies can be thought of as a GANTT diagram:

example gantt

In our case, we want to prevent residencies from being added/edited where the outcome would be any other residency having:

- otherResidency.moveIn >= newResidency.moveIn
AND
- otherResidency.moveOut <= newResidency.moveOut

Does that seem correct?

**Comment by shailee-m** [comment]
> Lets consider whether we should prevent **overlapping** residencies here, as opposed to just multiple _active_ residencies.

Understood. So no 2 residencies of resident can overlapping active durations.

**Comment by brylie** [comment]
Yeah, I think that is the issue we are trying to prevent. I will double check with Marjo to be sure.

**Comment by brylie** [comment]
Go ahead and write the code to account for both scenarios:

- overlapping active residency
- overlapping ended residency

I think both should be simple database queries, so don't need to be wrapped in any complicated logic.

File: app/server/methods/residencies.js [hunk 1]
```diff
@@ -81,5 +81,119 @@ Meteor.methods({
     }
 
     return Residencies.find(selector).map(residency => residency._id);
+  },
+
+  /*Check if the current resident has other residencies */
+  hasOtherActiveResidencies({residentId, residencyId, moveOut, moveIn}) {
+    console.log({residentId, residencyId, moveOut, moveIn},"jbdhfbdbhjdf")
```
Comment (brylie) [inline app/server/methods/residencies.js]: Remember to clean up console logs.

File: app/both/collections/residencies.js [hunk 2]
```diff
@@ -6,19 +6,35 @@ Residencies = new Mongo.Collection('residencies');
 var ResidenciesSchema = new SimpleSchema({
   residentId: {
     type: String,
+    custom() {
+      const residencyId = this.docId;
+      let currentRecord = this.obj
+        ? (
```
Comment (brylie) [inline app/both/collections/residencies.js]: Just add a small comment here describing why the logic is written as such.

**Comment by brylie** [comment]
This looks good. Just a small comment for clarification. Also, lets make sure to use Prettier on all files you are editing for now on.

https://prettier.io/

**Comment by brylie** [comment]
Great work!

**Comment by shailee-m** [comment]
@brylie  Thank you. Which next task can I pick?

**Comment by brylie** [comment]
@shailee-m can you verify this task? https://github.com/GeriLife/wellbeing/issues/445

**Pull request merged**